addOns/oast/src/main/java/org/zaproxy/addon/oast/services/interactsh/InteractshOptionsPanelTab.java

```diff
@@ -0,0 +1,225 @@
+    private ZapTextField authToken;
+    private ZapNumberSpinner pollingFrequencySpinner;
+    private JXTable payloadsTable;
+    private PayloadsTableModel payloadsTableModel;
+    private JButton newPayloadButton;
+
+    public InteractshOptionsPanelTab(InteractshService interactshService) {
+        super(interactshService.getName());
+
+        this.interactshService = interactshService;
+
+        int rowIndex = -1;
+
+        JLabel serverUrlLabel =
+                new JLabel(Constant.messages.getString("oast.interactsh.options.label.url"));
+        serverUrlLabel.setLabelFor(getServerUrl());
+        this.add(
+                serverUrlLabel, LayoutHelper.getGBC(0, ++rowIndex, 1, 0.4, new Insets(2, 2, 2, 2)));
+        this.add(getServerUrl(), LayoutHelper.getGBC(1, rowIndex, 1, 0.6, new Insets(2, 2, 2, 2)));
+
+        JLabel authTokenLabel =
+                new JLabel(Constant.messages.getString("oast.interactsh.options.label.authToken"));
+        authTokenLabel.setLabelFor(getAuthToken());
+        this.add(
+                authTokenLabel, LayoutHelper.getGBC(0, ++rowIndex, 1, 0.4, new Insets(2, 2, 2, 2)));
+        this.add(getAuthToken(), LayoutHelper.getGBC(1, rowIndex, 1, 0.6, new Insets(2, 2, 2, 2)));
+
+        JLabel pollingFrequencyLabel =
+                new JLabel(
+                        Constant.messages.getString(
+                                "oast.interactsh.options.label.pollingFrequency"));
+        pollingFrequencyLabel.setLabelFor(getPollingFrequencySpinner());
+        this.add(
+                pollingFrequencyLabel,
+                LayoutHelper.getGBC(0, ++rowIndex, 1, 0.4, new Insets(2, 2, 2, 2)));
+        this.add(
+                getPollingFrequencySpinner(),
+                LayoutHelper.getGBC(1, rowIndex, 1, 0.6, new Insets(2, 2, 2, 2)));
+
+        this.add(
+                getNewPayloadButton(),
+                LayoutHelper.getGBC(1, ++rowIndex, 1, 0, new Insets(2, 2, 2, 2)));
+
+        JScrollPane scrollPane = new JScrollPane(getPayloadsTable());
+        scrollPane.setBorder(
+                new TitledBorder(
+                        Constant.messages.getString(
+                                "oast.interactsh.options.label.activePayloads")));
+        this.add(
+                scrollPane,
+                LayoutHelper.getGBC(0, ++rowIndex, GridBagConstraints.REMAINDER, 1.0, 1.0));
+    }
+
+    ZapTextField getServerUrl() {
+        if (serverUrl == null) {
+            serverUrl = new ZapTextField();
+        }
+        return serverUrl;
+    }
+
+    ZapTextField getAuthToken() {
+        if (authToken == null) {
+            authToken = new ZapTextField();
+        }
+        return authToken;
+    }
+
+    ZapNumberSpinner getPollingFrequencySpinner() {
+        if (pollingFrequencySpinner == null) {
+            pollingFrequencySpinner =
+                    new ZapNumberSpinner(
+                            InteractshParam.MINIMUM_POLLING_FREQUENCY, 60, Integer.MAX_VALUE);
```

Comment:
I'm not sure if this is seconds or minutes, but max int seconds is like 68 years. I know there are arguments for not limiting people but....

Reply:
Agreed, we can have an upper limit... what do you suggest?

Reply:
I dunno an hour, 6 hours. Six even seems like a stretch to me, but I think that should be a ton of flexibility.